src/vcl/ImageData.cc

```diff
@@ -752,16 +775,28 @@ void ImageData::delete_object()
     /*      COPY FUNCTIONS      */
     /*  *********************** */
 
-void ImageData::copy_cv(const cv::Mat &cv_img)
+void ImageData::deep_copy_cv(const cv::Mat &cv_img)
+{
+    _channels = cv_img.channels();
+
+    _height = cv_img.rows;
+    _width  = cv_img.cols;
+
+    _cv_type = cv_img.type();
+
+    _cv_img = cv_img.clone(); // deep copy
+}
+
+void ImageData::shallow_copy_cv(const cv::Mat &cv_img)
 {
     _channels = cv_img.channels();
 
     _height = cv_img.rows;
-    _width = cv_img.cols;
+    _width  = cv_img.cols;
 
     _cv_type = cv_img.type();
 
-    _cv_img = cv_img.clone();
+    _cv_img = cv_img; // shallow copy
```

Comment:
Probably a silly question but is this still thread safe? That would be hard to catch with our tests.

Reply:
none of this is thread safe. these changes do not affect that.